--- /tmp/api_client.py	2024-02-25 20:40:28.143350042 +0100
+++ kubernetes_asyncio/client/api_client.py	2024-02-25 20:40:32.954201652 +0100
@@ -535,10 +535,13 @@
 
         content_types = [x.lower() for x in content_types]
 
-        if (method == 'PATCH' and
-                'application/json-patch+json' in content_types and
-                isinstance(body, list)):
-            return 'application/json-patch+json'
+        if method == 'PATCH':
+            if ('application/json-patch+json' in content_types and
+                    isinstance(body, list)):
+                return 'application/json-patch+json'
+            if ('application/strategic-merge-patch+json' in content_types and
+                    (isinstance(body, dict) or hasattr(body, "to_dict"))):
+                return 'application/strategic-merge-patch+json'
 
         if 'application/json' in content_types or '*/*' in content_types:
             return 'application/json'
