diff --git a/kubernetes_asyncio/client/configuration.py b/kubernetes_asyncio/client/configuration.py
index 720bb81f..b2522c23 100644
--- a/kubernetes_asyncio/client/configuration.py
+++ b/kubernetes_asyncio/client/configuration.py
@@ -12,6 +12,7 @@

 from __future__ import absolute_import

+import asyncio
 import copy
 import logging
 import sys
@@ -370,7 +371,7 @@ conf = client.Configuration(
         self.__logger_format = value
         self.logger_formatter = logging.Formatter(self.__logger_format)

-    def get_api_key_with_prefix(self, identifier, alias=None):
+    async def get_api_key_with_prefix(self, identifier, alias=None):
         """Gets API key (with prefix if set).

         :param identifier: The identifier of apiKey.
@@ -378,7 +379,9 @@ conf = client.Configuration(
         :return: The token for api key authentication.
         """
         if self.refresh_api_key_hook is not None:
-            self.refresh_api_key_hook(self)
+            result = self.refresh_api_key_hook(self)
+            if asyncio.iscoroutine(result):
+                await result
         key = self.api_key.get(identifier, self.api_key.get(alias) if alias is not None else None)
         if key:
             prefix = self.api_key_prefix.get(identifier)
@@ -402,7 +405,7 @@ conf = client.Configuration(
             basic_auth=username + ':' + password
         ).get('authorization')

-    def auth_settings(self):
+    async def auth_settings(self):
         """Gets Auth Settings dict for api client.

         :return: The Auth Settings information dict.
@@ -413,7 +416,7 @@ conf = client.Configuration(
                 'type': 'api_key',
                 'in': 'header',
                 'key': 'authorization',
-                'value': self.get_api_key_with_prefix(
+                'value': await self.get_api_key_with_prefix(
                     'BearerToken',
                 ),
             }
