--- /tmp/configuration.py	2024-01-15 23:55:23.361535239 +0100
+++ kubernetes_asyncio/client/configuration.py	2024-01-15 23:55:30.793395231 +0100
@@ -189,6 +189,10 @@
         self.assert_hostname = None
         """Set this to True/False to enable/disable SSL hostname verification.
         """
+        self.tls_server_name = None
+        """SSL/TLS Server Name Indication (SNI)
+           Set this to the SNI value expected by Kubernetes API.
+        """
 
         self.connection_pool_maxsize = 100
         """This value is passed to the aiohttp to limit simultaneous connections.
