diff --git a/kubernetes_asyncio/client/rest.py b/kubernetes_asyncio/client/rest.py
index ce6c8734..8fa35a94 100644
--- a/kubernetes_asyncio/client/rest.py
+++ b/kubernetes_asyncio/client/rest.py
@@ -71,7 +71,14 @@ class RESTClientObject(object):
         # https pool manager
         self.pool_manager = aiohttp.ClientSession(
             connector=connector,
-            trust_env=True
+            trust_env=True,
+            # Watch events containing large resource objects can exceed
+            # aiohttp's default read buffer size.
+            #
+            # There is no hard-limit defined by k8s, but the etcd default
+            # maximum request size is 1.5MiB.
+            # https://github.com/kubernetes/kubernetes/issues/19781
+            read_bufsize=2**21
         )
 
     async def close(self):
