diff --git a/kubernetes/client/rest.py b/kubernetes/client/rest.py
index 3731698f0..a3c7257d8 100644
--- a/kubernetes/client/rest.py
+++ b/kubernetes/client/rest.py
@@ -148,6 +148,10 @@ class RESTClientObject(object):
                 if query_params:
                     url += '?' + urlencode(query_params)
                 if ('Content-Type' not in headers) or (re.search('json', headers['Content-Type'], re.IGNORECASE)):
+                    if ('Content-Type' in headers) and (headers['Content-Type'] == 'application/json-patch+json'):
+                        if not isinstance(body, list):
+                            headers['Content-Type'] = \
+                                'application/strategic-merge-patch+json'
                     request_body = None
                     if body is not None:
                         request_body = json.dumps(body)
