diff --git a/kubernetes/client/configuration.py b/kubernetes/client/configuration.py
index e92e0a653..939548d51 100644
--- a/kubernetes/client/configuration.py
+++ b/kubernetes/client/configuration.py
@@ -191,6 +191,10 @@ conf = client.Configuration(
         self.assert_hostname = None
         """Set this to True/False to enable/disable SSL hostname verification.
         """
+        self.tls_server_name = None
+        """SSL/TLS Server Name Indication (SNI)
+           Set this to the SNI value expected by the server.
+        """
 
         self.connection_pool_maxsize = multiprocessing.cpu_count() * 5
         """urllib3 connection pool's maximum number of connections saved
diff --git a/kubernetes/client/rest.py b/kubernetes/client/rest.py
index f805d0661..06a9a3564 100644
--- a/kubernetes/client/rest.py
+++ b/kubernetes/client/rest.py
@@ -72,6 +72,9 @@ class RESTClientObject(object):
         if configuration.socket_options is not None:
             addition_pool_args['socket_options'] = configuration.socket_options
 
+        if configuration.tls_server_name:
+            addition_pool_args['server_hostname'] = configuration.tls_server_name
+
         if maxsize is None:
             if configuration.connection_pool_maxsize is not None:
                 maxsize = configuration.connection_pool_maxsize
